src/dev/formatWhitespace.m

```diff
@@ -0,0 +1,66 @@
+function formatWhitespace()
+    % FORMATWHITESPACE Formats whitespace in the active document
+    %
+    %   Rules:
+    %       - Smart-indent
+    %       - Indent content of all function with 4 spaces per level
+    %       - Remove whitespace in empty lines
+    %       - Preserve indentantion after line continuations, i.e. ...
+
+    import matlab.desktop.editor.*
+    try
+        d        = getActive();
+        oldLines = textToLines(d.Text);
+    catch
+        error('formatWhitespace:noEditorApi','Check that the Editor API is available.')
+    end
+
+    % Parse file into a tree
+    tree = mtree(d.Text);
+
+    % Check for mlint error
+    if ~isempty(tree.mtfind('Kind','ERR'))
+        [~,name,ext] = fileparts(obj.Filename);
+        error('formatWhitespace:mlintError','''%s'' contains syntax errors. Cannot proceed.',[name,ext])
+    end
+
+    % Smart indent except for crafted continuations of line
+    d.smartIndentContents()
+    lines      = textToLines(d.Text);
+    iDots      = ~cellfun('isempty',regexp(lines, '\.\.\.','once'));
```

Comment:
Wouldn't this be more readable using just `strfind` (I thought there was also a MATLAB function to convert escape literal strings into regular expressions, but I can't find it back).